pkg/cli/params.go

```diff
@@ -63,6 +66,53 @@ func (p *TektonParams) Clientset() (versioned.Interface, error) {
 	return p.clientset, nil
 }
 
+func (p *TektonParams) KubeClient() (k8s.Interface, error) {
+	if p.kubeClient != nil {
+		return p.kubeClient, nil
+	}
+
+	kcPath, err := resolveKubeConfigPath(p.kubeConfigPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to resolve kubeconfig path")
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kcPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "parsing kubeconfig failed")
+	}
+
+	k8scs, err := k8s.NewForConfig(config)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create ks8 client from config %s", err)
+	}
+
+	p.kubeClient = k8scs
+	return p.kubeClient, nil
+}
+
+func (p *TektonParams) Clients() (*Clients, error) {
```

Comment:
There is too much duplicated code here, no need to do the `resolveKubeConfigPath` & co multplie times :angel: If you go for having only `Clients`, you can do something like the following

```go
func (p *TektonParams) Clients() (*Clients, error) {
	if p.clients != nil {
		return p.clients, nil
	}
	kcPath, err := resolveKubeConfigPath(p.kubeConfigPath)
	if err != nil {
		return nil, errors.Wrap(err, "failed to resolve kubeconfig path")
	}
	config, err := clientcmd.BuildConfigFromFlags("", kcPath)
	if err != nil {
		return nil, errors.Wrapf(err, "Parsing kubeconfig failed")
	}
	cs, err := versioned.NewForConfig(config)
	if err != nil {
		return nil, err
	}
	kcs, err := corev1.NewForConfig(config)
	if err != nil {
		return nil, err
	}
	clients := &Client{
		Tekton: cs,
		Kube:   kcs,
	}
	p.clients = clients
	return p.clients, nil
}
```

Reply:
:sweat_smile: resolved